src/apps/btc/btc_sign.c

```diff
@@ -200,15 +201,26 @@ static bool _is_valid_keypath(
                 script_config->config.simple_type, keypath, keypath_count, expected_bip44_coin)) {
             return false;
         }
-        const uint32_t account = keypath[2];
-        const uint32_t change = keypath[3];
-        if (account != expected_bip44_account || (must_be_change && change != 1)) {
+        break;
+    default:
+        return false;
+    }
+
+    // check that keypath_account is a prefix to keypath with two elements left (change, address).
```

Comment:
Are you changing this again for multisig's extra non-hardened derivation level?

Reply:
which keypath format are you talking about? electrum has bip48 which is an extra hardened derivation level, so the account keypath is four elements; m/48'/coin'/acount'/script_type' (script_type being 2 for p2wsh).

Reply:
Never mind, they are both hardened.

Reply:
No, I was right before, [BIP45](https://github.com/bitcoin/bips/blob/master/bip-0045.mediawiki) uses this weird format : `m/purpose'/cosigner_index/change/address_index`. But we can choose to just not support it.

Reply:
BIP48 does not exist btw, it's all super confusing. It's only documentation is here: https://github.com/bitpay/copay#wallet-export-format , which quotes:

> Since version 1.5, Copay uses the root m/48' for hardware multisignature wallets. This was coordinated with Ledger and Trezor teams. While the derivation path format is still similar to BIP44, the root was in order to indicate that these wallets are not discoverable by scanning addresses for funds. Address generation for multisignature wallets requires the other copayers extended public keys.

...meaning the checks are fine as they are.

Reply:
> No, I was right before, [BIP45](https://github.com/bitcoin/bips/blob/master/bip-0045.mediawiki) uses this weird format : `m/purpose'/cosigner_index/change/address_index`. But we can choose to just not support it.

yeah, as we've done with the other keypaths, I prefer to be strict at first and whitelist more if the need arises. Thanks for checking the other formats.